### DISCUSSION TITLE
Error: Mapping <middleware> not found. Maybe you provided a wrong middlewarePath? Available paths: <another-middleware>

### DISCUSSION THREAD

**Discussion by btakita** [vercel/vercel#8628] [Help]
I'm trying to debug

```
Error: Mapping render not found. Maybe you provided a wrong middlewarePath? Available paths: api
```

This vercel cli repo has been helpful in debugging another issue I had with the cli. Is the software that emits this error message available as open source software? It will help in debugging specific issues.

Re: this specific issue, I have an `functions/api.func` & `functions/render.func` with each having a `.vc-config.json` file.

`functions/api.func/.vc-config.json`
```json
{
  "runtime": "edge",
  "entrypoint": "index.js"
}
```

`functions/render.func/.vc-config.json`
```json
{
  "runtime": "nodejs16.x",
  "handler": "index.js",
  "launcherType": "Nodejs"
}
```

`vercel.json`
```json
{
	"alias": ["website.com"],
	"cleanUrls": true,
	"github": {
		"enabled": true
	},
	"functions": {
		"api/**/*.js": {
			"memory": 1024,
			"maxDuration": 10
		},
		"render/**/*.js": {
			"memory": 1024,
			"maxDuration": 10
		}
	}
}
```

https://github.com/vercel/next.js/issues/38308

**Comment by amyegan** [comment]
Thanks for reaching out @btakita! The source code for that error is not publicly available. Your config looks right to me for a Build Output API [Serverless Function](url). 

Since it's saying that `functions/render.func` could not be found, but `functions/api.func` is found, I'm curious about your file structure. Is `index.js` the correct file, or should it be set to something else in `functions/render.func/.vc-config.json`? Are you seeing any other errors or warnings during build (including any that seem unrelated)? 

If you are able to share deployment links or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to demonstrate the problem, I'm happy to help with debugging.
